pyclaragenomics/samples/sample_cudapoa

```diff
@@ -35,41 +35,67 @@ def run_cudapoa(groups, msa, print_output):
     """
     # Get avaialble memory information
     free, total = cuda.cuda_get_mem_info(cuda.cuda_get_device())
+    gpu_mem_per_batch = 0.9 * free
+
+    # Calculate max bounds for sequence size and sequences per POA.
+    max_sequences_per_poa = 0
+    max_seq_sz = 0
+    for group in groups:
+        longest_seq = len(max(group, key=len))
+        max_seq_sz = longest_seq if longest_seq > max_seq_sz else max_seq_sz
+        seq_in_poa = len(group)
+        max_sequences_per_poa = seq_in_poa if seq_in_poa > max_sequences_per_poa else max_sequences_per_poa
+
+    max_sequences_per_poa += 1
+    max_seq_sz += 1
 
     # Create batch
-    max_sequences_per_poa = 100
-    max_seq_sz = 1024
-    gpu_mem_per_batch = 0.9 * free
     batch = cudapoa.CudaPoaBatch(max_sequences_per_poa,
                                  max_seq_sz,
                                  gpu_mem_per_batch,
                                  stream=None,
-                                 output_type=("msa" if msa else "consensus"))
+                                 output_type=("msa" if msa else "consensus"),
+                                 banded=True)
 
     # Add poa groups to batch
     initial_count = 0
-    for i, group in enumerate(groups):
+    poa_index = 0
+    while (poa_index < len(groups)):
+        group = groups[poa_index]
+
         group_status, seq_status = batch.add_poa_group(group)
-        if (group_status == 0):
-            pass
 
-        # Once batch is full, run POA processing
-        if ((group_status == 1) or (i == len(groups) - 1)):
+        # If group was added and more space is left in batch, continue onto next group.
+        if (group_status == 0):
+            for seq_index, status in enumerate(seq_status):
+                if status != 0:
+                    print("Could not add sequence {} to POA {} - error {}".format(seq_index, poa_index, cudapoa.status_to_str(status)))
+            poa_index += 1
+
+        # Otherwise print error.
+        # Once batch is full or no groups are left, run POA processing.
+        if ((group_status == 1) or (poa_index == len(groups) - 1)):
```

Comment:
do we need to consider a scenario that a single POA group is too large to be added? e.g. in bonito the first window consists of reads with length near 14K. Using full-alignment on a GPU with small memory it can run out of memory. Then will it get stuck in an infinite while loop? this can be tested by changing `gpu_mem_per_batch = 0.9 * free` to something smaller than 2GB.

Reply:
Good catch! In this implementation it happened only when the last group was not addable to the batch, so it got stuck in an infinite loop.

Also found a deeper bug in the binding that was always enabling banded for cudapoa. Fixed that as well.